### PR TITLE
Master onboarding hr expense dli

### PR DESCRIPTION
**Pull request by dli-odoo**
task : https://www.odoo.com/web#id=29303&view_type=form&model=project.task&action=333&active_id=131&menu_id=4720

pad : https://pad.odoo.com/p/r.5ed31ae59070679bfc6ab9799bbf35d5

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/hr_expense/data/hr_expense_demo.xml [hunk 1]
```diff
@@ -67,7 +67,7 @@
         </record>
 
         <record id="hotel_bill_expense" model="hr.expense">
-            <field name="name">Hotel Expenses</field>
+            <field name="name">Hotel Expenses Bill</field>
```
Comment (atp-odoo) [inline addons/hr_expense/data/hr_expense_demo.xml]: Remove unused change.

File: addons/hr_expense/models/hr_expense.py [hunk 2]
```diff
@@ -104,17 +104,15 @@ def submit_expenses(self):
...
-            'view_mode': 'form',
-            'res_model': 'hr.expense.sheet',
-            'target': 'current',
-            'context': {
-                'default_expense_line_ids': [line.id for line in self],
-                'default_employee_id': self[0].employee_id.id,
-                'default_name': self[0].name if len(self.ids) == 1 else ''
-            }
+        vals = {
+            'name': self.name,
```
Comment (atp-odoo) [inline addons/hr_expense/models/hr_expense.py]: oops!! here you got multiple records. so you should iterate record.

File: addons/hr_expense/views/hr_expense_views.xml [hunk 3]
```diff
@@ -60,7 +60,7 @@
                         </group><group>
                             <field name="reference"/>
                             <field name="date"/>
-                            <field name="account_id"/>
+                            <field name="account_id" groups="account.group_account_manager"/>
```
Comment (atp-odoo) [inline addons/hr_expense/views/hr_expense_views.xml]: it should be : account.group_account_user for accountant..

File: addons/sale_expense/models/hr_expense.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import re
```
Comment (atp-odoo) [inline addons/sale_expense/models/hr_expense.py]: unused import

File: addons/sale_expense/models/hr_expense.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+from odoo import api, fields, models, _
```
Comment (atp-odoo) [inline addons/sale_expense/models/hr_expense.py]: unused import api and _ . remove it.

File: addons/sale_expense/models/hr_expense.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+from odoo.exceptions import UserError
+
+class HrExpenseInherit(models.Model):
```
Comment (atp-odoo) [inline addons/sale_expense/models/hr_expense.py]: class name should be "HrExpense"

File: addons/sale_expense/models/sale_order.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import re
```
Comment (atp-odoo) [inline addons/sale_expense/models/sale_order.py]: why every where import re? remove it unused.

File: addons/sale_expense/models/sale_order.py [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+class SaleOrderInherit(models.Model):
```
Comment (atp-odoo) [inline addons/sale_expense/models/sale_order.py]: class name should be SaleOrder

File: addons/sale_expense/views/hr_expense_views.xml [hunk 9]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+	<record model="ir.ui.view" id="hr_expense_inherit_sale_expense">
```
Comment (atp-odoo) [inline addons/sale_expense/views/hr_expense_views.xml]: correct indentation.!!

File: addons/hr_expense/static/src/js/tour.js [hunk 10]
```diff
@@ -6,34 +6,58 @@ var tour = require('web_tour.tour');
...
     trigger: '.o_app[data-menu-xmlid="hr_expense.menu_hr_expense_root"], .oe_menu_toggler[data-menu-xmlid="hr_expense.menu_hr_expense_root"]',
-    content: _t("Want to manage your employee expenses and receipts? <i>Start here</i>."),
+    content: _t("Go to the expense to attach a receipt."),
     position: 'bottom',
 }, {
-    trigger: 'a#o_mail_test',
-    content: _t("Click to try <b>submitting an expense by email</b>. You can attach a photo of the receipt to the mail."),
+    trigger: '.o_list_button_add',
+    extra_trigger: ".o_expense_tree",
+    content: _t("<p> List your Expense by Creating it here.</p>"),
```
Comment (bst-odoo) [inline addons/hr_expense/static/src/js/tour.js]: This tooltip doesn't mean anything

**Comment by hmo-odoo** [comment]
merged at https://github.com/odoo/odoo/commit/406ba5dcc07076c434751153b62b8891998d0f50

**Pull request closed without merging**